src/main/java/seedu/duke/data/ModuleList.java

```diff
@@ -73,6 +75,10 @@ public void reset() {
      * @return true if a module in the module list already has that module code, or false otherwise.
      */
     public boolean isModuleExists(String moduleCode) {
-        return (getModule(moduleCode) != null);
+        try {
+            return list.contains(getModule(moduleCode));
+        } catch (NoSuchModuleException e) {
+            return false;
+        }
```

Comment:
Any suggestions on a better implementation here?

Reply:
I think this could work?
```
try {
    getModule(moduleCode);
    return true;
} catch (...) {
    return false;
}